runtime/jcl/common/system.c

```diff
@@ -70,12 +70,34 @@ jstring JNICALL Java_java_lang_System_getEncoding(JNIEnv *env, jclass clazz, jin
 	return getEncoding(env, encodingType);
 }
 
+#if JAVA_SPEC_VERSION >= 18
+jstring JNICALL Java_java_lang_System_getSysPropBeforePropertiesInitialized (JNIEnv *env, jclass clazz, jint sysPropID)
+{
+	const char *sysPropValue = NULL;
+	jstring result = NULL;
+	PORT_ACCESS_FROM_ENV(env);
+
+	switch (sysPropID) {
+	case 0:    /* os.version */
+		sysPropValue = j9sysinfo_get_OS_version();
```

Comment:
This isn't the same as the original code which checks if the sysinfo function returns NULL and if so sets the result to "unknown". The spec for j9sysinfo_get_OS_version does contain "return ... , NULL on error".

Reply:
Set `sysPropValue` to `unknown` when `j9sysinfo_get_OS_version` returns `NULL`.